hbw/app/models/hbw/common/adapter.rb

```diff
@@ -104,17 +110,12 @@ def claim_task(email, task_id)
       end
 
       def form(task_id, entity_class)
-        task = task_for_email_and_task_id(task_id, entity_class)
         HBW::Form.with_connection(api) do
-          HBW::Form.fetch(task, entity_class)
-        end
-      end
+          form = HBW::Form.fetch(task_id, entity_class)
 
-      def form_definition(task_id, entity_class)
-        task = HBW::Task.new('id' => task_id, 'processDefinition' => nil)
+          form.fetch_fields!
```

Comment:
Maybe you should fetch fields always in `fetch`? Do we really need form without fields data?